src/core_init_atmosphere/mpas_kd_tree.F

```diff
@@ -0,0 +1,394 @@
+module mpas_kd_tree
+
+   !***********************************************************************
+   !
+   !  module mpas_kd_tree
+   !
+   !> \brief   MPAS KD-Tree module
+   !> \author  Miles A. Curry
+   !> \date    01/28/20
+   !> A KD-Tree implementation to create and search perfectly balanced
+   !> KD-Trees.
+   !>
+   !> Use `mpas_kd_type` dervied type to construct points for mpas_kd_construct:
+   !>
+   !> real, dimension(:,:), allocatable :: array
+   !> type (mpas_kd_tree), pointer :: tree => null()
+   !> type (mpas_kd_tree), dimension(:), pointer :: points => null()
+   !>
+   !> allocate(array(k,n)) ! K dims and n points
+   !> allocate(points(n))
+   !> array(:,:) = (/.../)  ! Fill array with values
+   !>
+   !> do i = 0, n
+   !>    allocate(points(i) % point(k))    ! Allocate point with k dimensions
+   !>    points(i) % point(:) = array(:,i)
+   !>    points(i) % id = i                ! Or a value of your choice
+   !> enddo
+   !>
+   !> tree => mpas_kd_construct(points, k)
+   !
+   !-----------------------------------------------------------------------
+   use mpas_derived_types
```

Comment:
Rather than using the entire derived_types module, we can just
```fortran
   use mpas_kind_types, only : RKIND
```

Reply:
Sorry, I should have emphasized the change from `mpas_derived_types` to `mpas_kind_types` in addition to the use of an `only` clause.